src/engine/move_handling.h

```diff
@@ -246,6 +248,15 @@ constexpr BitBoard performMove(const BitBoard& board, const movegen::Move& move,
     return newBoard;
 }
 
+// Helper: using inside loops means redundant colour checks
+constexpr BitBoard performMove(const BitBoard& board, const movegen::Move& move, uint64_t& hash, Player player)
+{
+    if (player == PlayerWhite)
+        return performMove<PlayerWhite>(board, move, hash);
+    else
+        return performMove<PlayerBlack>(board, move, hash);
+}
```

Comment:
This could maybe be a cool way to implement it without adding redundancy? It will have same signature, but without the template, so you can branch at compile time using the template or runtime when not :)

Maybe it's less explicit? 

```suggestion
constexpr BitBoard performMove(const BitBoard& board, const movegen::Move& move, uint64_t& hash)
{
    if (board.player == PlayerWhite)
        return performMove<PlayerWhite>(board, move, hash);
    else
        return performMove<PlayerBlack>(board, move, hash);
}
```